app/controllers/concerns/course/users_controller_management_concern.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+module Course::UsersControllerManagementConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :authorize_show!, only: [:students, :staff, :requests, :invitations]
+    before_action :authorize_edit!, only: [:update, :destroy]
+  end
+
+  def update # :nodoc:
+    if update_course_user(course_user_params)
+      success = t('course.users.update.success', role: t("course.users.role.#{@course_user.role}"))
```

Comment:
That seems like the best available, and seems to be as documented in the ActiveRecord translation section of the Rails guides.